### DISCUSSION TITLE
python程式問題

### DISCUSSION THREAD

**Discussion by cordydeng** [1101-GS4719-C/Python-Community-GS4719-C#375] [Q&A] [answered]
想請問一下python內要怎麼在單行連續輸入，如果要一次input很多值的話好像只能換行輸入

**Comment by awkward-willy** [comment, marked as answer]
若要分別存入不同的變數中，可以利用 input 的 split，預設是以空格分割，各輸入值會以 str 儲存
如：str1, str2 = input().split()
若要針對每項輸入值做特定function轉換，也可以搭配map()
如：num1, num2 = map(int,input().split())
若要存入 list 
li = list(map(int, input().split()))

**Reply by cordydeng** [reply]
謝謝大神的指導，之前用map時發現只能用於同種型態，對於不同型態的變數就沒辦法了

**Comment by handsomeniko** [comment]
可以用分號;來做單行輸入不同資料
例如:a=1;b=2;c=3

**Comment by pekoranoinu** [comment]
原來python可以在單行連續輸入，我之前都是換行慢慢來，真是學到新知了。

**Reply by cordydeng** [reply]
每次換行的話版面會變超級醜 , 學到這招以後就不用看到雜亂的版面了

**Comment by chupochun** [comment]
我以前也是一行一行慢慢打 謝謝提問與回答

**Comment by ghost** [comment]
補充二F，若是想用特殊字元做為切割，可在split的sep參數調整
如下方以,作為分割
str1, str2 = input().split(sep=',')

**Comment by Ben6uo** [comment]
雖然直接用' ; '或是直接換行的效果是一樣,但是在掛板上卻方便了很多 感謝大大的答案
